exporter/awsprometheusremotewriteexporter/factory.go

```diff
@@ -0,0 +1,120 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsprometheusremotewriteexporter
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	prw "go.opentelemetry.io/collector/exporter/prometheusremotewriteexporter"
+)
+
+const (
+	typeStr = "awsprometheusremotewrite" // The value of "type" key in configuration.
+)
+
+type AwsFactory struct {
+	component.ExporterFactory
+}
+
+// NewFactory returns a factory of the AWS Prometheus Remote Write exporter that can be registered to the Collector.
+func NewFactory() component.ExporterFactory {
+	return &AwsFactory{ExporterFactory: prw.NewFactory()}
+}
+
+func (af *AwsFactory) Type() configmodels.Type {
+	return typeStr
+}
+
+func (af *AwsFactory) CreateMetricsExporter(ctx context.Context, params component.ExporterCreateParams,
+	cfg configmodels.Exporter) (component.MetricsExporter, error) {
+	prwCfg := cfg.(*Config)
+
+	prwCfg.HTTPClientSettings.CustomRoundTripper = func(next http.RoundTripper) (http.RoundTripper, error) {
+		if !validateAuthConfig(prwCfg.AuthSettings) {
+			return nil, errors.New("invalid authentication configuration")
+		}
+
+		return newSigningRoundTripper(prwCfg.AuthSettings, next)
+	}
+
+	client, cerr := prwCfg.HTTPClientSettings.ToClient()
+	if cerr != nil {
+		return nil, cerr
+	}
+
+	// initialize an upstream exporter and pass it an http.Client with interceptor
+	prwe, err := prw.NewPrwExporter(prwCfg.Namespace, prwCfg.HTTPClientSettings.Endpoint, client, prwCfg.ExternalLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	prwexp, err := exporterhelper.NewMetricsExporter(
+		cfg,
+		params.Logger,
+		prwe.PushMetrics,
+		exporterhelper.WithTimeout(prwCfg.TimeoutSettings),
+		exporterhelper.WithQueue(prwCfg.QueueSettings),
+		exporterhelper.WithRetry(prwCfg.RetrySettings),
+		exporterhelper.WithShutdown(prwe.Shutdown),
+	)
+
+	return prwexp, err
+}
+
+func (af *AwsFactory) CreateDefaultConfig() configmodels.Exporter {
+	qs := exporterhelper.CreateDefaultQueueSettings()
+	qs.Enabled = false
+
+	ts := exporterhelper.CreateDefaultRetrySettings()
+	ts.Enabled = false
+
+	cfg := &Config{
+		Config: prw.Config{
+			ExporterSettings: configmodels.ExporterSettings{
+				TypeVal: typeStr,
+				NameVal: typeStr,
+			},
+			Namespace:       "",
+			ExternalLabels:  map[string]string{},
+			TimeoutSettings: exporterhelper.CreateDefaultTimeoutSettings(),
+			RetrySettings:   ts,
+			QueueSettings:   qs,
+			HTTPClientSettings: confighttp.HTTPClientSettings{
+				Endpoint: "http://some.url:9411/api/prom/push",
```

Comment:
seems weird to set a default value that must always be overridden? Like this value can never be used right?

Reply:
yea this will never be used. It's just the design of the collector to first create a default config and then overwrite it with the user configuration. I guess it avoids having null fields which could cause problems if not checked for in the rest of the code

Reply:
If the user does not add their own endpoint, will the plugin fail initialization?

As I said before... I actually don't work closely with the collector, but in other similar agents, if a config option is not specified by a user and is null then it fails initialization, and that's what you want. Fail fast is good for users.

Reply:
in this case initialization does not fail, it fails when the http request is attempted. Good point, I will change this so that it fails initialization